bson/src/main/org/bson/codecs/pojo/PojoIdExtendingBsonWriter.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.bson.codecs.pojo;
+
+import org.bson.BsonBinary;
+import org.bson.BsonDbPointer;
+import org.bson.BsonReader;
+import org.bson.BsonRegularExpression;
+import org.bson.BsonTimestamp;
+import org.bson.BsonValue;
+import org.bson.BsonWriter;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+
+class PojoIdExtendingBsonWriter implements BsonWriter {
+    private static final CodecRegistry REGISTRY = fromProviders(new BsonValueCodecProvider());
+    private static final EncoderContext ENCODER_CONTEXT = EncoderContext.builder().build();
+    private static final String ID_NAME = "_id";
+    private final BsonValue id;
+    private final BsonWriter writer;
+
+
+    PojoIdExtendingBsonWriter(final BsonValue id, final BsonWriter writer) {
+        this.id = id;
+        this.writer = writer;
+    }
+
+    @Override
+    public void writeStartDocument() {
+        writer.writeStartDocument();
+        writer.writeName(ID_NAME);
```

Comment:
This will write the _id field at every level in the case where there are nested documents.  To avoid that it has to track what level it's at.

Have a look at com.mongodb.internal.connection.ElementExtendingBsonWriter.  I think it's worth re-using that class here, though you'll have to extend its behavior to support writing extra elements to the beginning of the document instead of the end.